clientgenv2/query.go

```diff
@@ -60,14 +60,14 @@ func fragmentsInOperationDefinition(operation *ast.OperationDefinition) ast.Frag
 }
 
 func fragmentsUnique(fragments ast.FragmentDefinitionList) ast.FragmentDefinitionList {
-	uniqueMap := make(map[string]*ast.FragmentDefinition)
+	seenFragments := make(map[string]struct{}, len(fragments))
+	uniqueFragments := make(ast.FragmentDefinitionList, 0, len(fragments))
 	for _, fragment := range fragments {
-		uniqueMap[fragment.Name] = fragment
-	}
-
-	uniqueFragments := make(ast.FragmentDefinitionList, 0, len(uniqueMap))
-	for _, fragment := range uniqueMap {
+		if _, seen := seenFragments[fragment.Name]; seen {
```

Comment:
I like this one better.
```suggestion
		if _, ok := seenFragments[fragment.Name]; ok {
```

Reply:
Looks good 🙂